### PR TITLE
Service cleanup

### PR DESCRIPTION
**Pull request by IEvangelist**
Encapsulate the WebAssembly specific bits into a service abstraction, that allows the MAUI bits to be independently implemented/ignored without throwing exception, or handling try/catch.

### CODE REVIEW

**Review by LittleLittleCloud — DISMISSED** [review]

File: app/SharedWebComponents/Pages/VoiceChat.razor.cs [hunk 1]
```diff
@@ -79,28 +79,11 @@ private void OnSendPrompt()
...
-                            // alternate API is something MAUI can implement.
 
-                            SpeechSynthesis.Speak(utterance);
+                        SpeechSynthesis.Speak(utterance);
 
-                            _isReadingResponse = false;
-                            StateHasChanged();
-                        }
+                        _isReadingResponse = false;
+                        StateHasChanged();
```
Comment (mattleibow) [inline app/SharedWebComponents/Pages/VoiceChat.razor.cs:86]: Should this rather be after an await or some API? This looks to immediately start the speech and then say it is done. Before this was in a callback/delegate. Maybe MAUI did it wrong and use the async void so as not to change the code?

**Review by mattleibow — APPROVED** [review]
Everything seems to be working :)

File: app/SharedWebComponents/Pages/VoiceChat.razor.cs (hunk 1 above)
Comment (IEvangelist) [reply app/SharedWebComponents/Pages/VoiceChat.razor.cs:86]: Oh, you're right. The callback provided the `duration`. I think this should be reverted.

File: app/maui-blazor/Services/MauiSpeechSynthesisService.cs [hunk 2]
```diff
@@ -40,35 +47,45 @@ public void Resume()
...
-            Locale = locale
-        };
-        await TextToSpeech.Default.SpeakAsync(utterance.Text, options, _cts.Token);
+            await textToSpeech.SpeakAsync(utterance.Text, options, _cts.Token);
+        }, _cts.Token);
+    }
+
+    public void UnsubscribeFromAvailableVoicesChanged()
+    {
+
```
Comment (IEvangelist) [inline app/maui-blazor/Services/MauiSpeechSynthesisService.cs:89]:
```suggestion
```

**Review by mattleibow — APPROVED** [review]

**Review by LittleLittleCloud — APPROVED** [review]

**Pull request merged**